spec/classes/mod/jk_spec.rb

```diff
@@ -115,7 +115,7 @@
 
         it_behaves_like 'minimal resources'
         it do
-          is_expected.to contain_file("/etc/httpd/conf.d/jk.conf")
+          is_expected.to contain_file("#{:mod_dir}/jk.conf")
```

Comment:
this is a bit of a gotcha. `contain_file` is referencing the file resource in the catalog by its title. If you look in [jk.pp](https://github.com/puppetlabs/puppetlabs-apache/blob/master/manifests/mod/jk.pp#L83) you'll see that the config file has a hard-coded title of 'jk.conf' which is why this test wasn't working. I think what you're looking for is something more like this:
```ruby
is_expected.to contain_file('jk.conf').with({ :path => "#{mod_dir}/jk.conf" })
```